base/range.jl

```diff
@@ -860,13 +860,19 @@ end
 
 median(r::Range{<:Real}) = mean(r)
 
-function in(x, r::Range)
+function in(x::Real, r::Range{<:Real})
+    n = step(r) == 0 ? 1 : round(Integer,(x-first(r))/step(r))+1
```

Comment:
A single implementation in a helper function strikes me as less complex :). Moreover, guards against changes being made in one place but not the other.